test/model/create.test.ts

```diff
@@ -29,3 +30,141 @@ test('creates a new entity with initial values', () => {
   })
   expect(exactUser).toHaveProperty('id', 'abc-123')
 })
+
+test('creates a new entity with initial values, including arrays', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      arrayProp: Array,
+    },
+  })
+
+  const exactUser = db.user.create({
+    id: 'abc-123',
+    arrayProp: [1, 2, 3],
+  })
+  expect(exactUser).toHaveProperty('id', 'abc-123')
+  expect(exactUser).toHaveProperty('arrayProp', [1, 2, 3])
+})
+
+test('supports nested objects in the model definition', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      name: String,
+      info: {
+        firstName: String,
+        lastName: String,
+        address: {
+          street: () => 'Yellow Brick Road',
+          number: () => 1,
+        },
+        tags: Array,
+      },
+    },
+  })
+
+  // Entity can be given exact values to seed.
+  const exactUser = db.user.create({
+    id: 'abc-123',
+    name: 'sampleUser',
+    info: {
+      firstName: 'Reginald',
+      lastName: 'Dwight',
+      address: {
+        number: 73,
+      },
+      tags: ['one', 'two'],
+    },
+  })
+
+  expect(exactUser).toHaveProperty('id', 'abc-123')
+  expect(exactUser).toHaveProperty('name', 'sampleUser')
+  expect(exactUser).toHaveProperty('info', {
+    firstName: 'Reginald',
+    lastName: 'Dwight',
+    address: {
+      street: 'Yellow Brick Road',
+      number: 73,
+    },
+    tags: ['one', 'two'],
+  })
+})
+
+test('relational properties can be declared in nested objects', () => {
+  const db = factory({
+    user: {
+      id: primaryKey(datatype.uuid),
+      name: String,
+      info: {
+        country: oneOf('country'),
+        firstName: String,
+        lastName: String,
+      },
+    },
+    country: {
+      id: primaryKey(datatype.uuid),
+      name: String,
+    },
+  })
+
+  const japan = db.country.create({
+    name: 'Japan',
+  })
+
+  const exactUser = db.user.create({
+    name: 'user',
+    info: {
+      country: japan,
+      firstName: 'Ryuichi',
+      lastName: 'Sakamoto',
+    },
+  })
+
+  expect(exactUser).toHaveProperty('name', 'user')
+  expect(exactUser.info).toHaveProperty('firstName', 'Ryuichi')
+  expect(exactUser.info).toHaveProperty('lastName', 'Sakamoto')
+  expect(exactUser.info.country).toHaveProperty('name', 'Japan')
+})
+
+test('throws an exception when primary key is not set at root level', () => {
```

Comment:
May I please ask you to move any primaryKey-related test cases to `test/primaryKey.test.ts`?

It already contains some tests for primary key exceptions.